magic-methods/code_stego/code_stego/encoder.py

```diff
@@ -0,0 +1,94 @@
+"""Implements the main encoder function, and its helper functions"""
+
+from io import BytesIO
+from typing import Sequence
+
+import numpy as np
+import PIL.Image
+from pygments import highlight
+from pygments.formatters import get_formatter_by_name
+from pygments.lexers import get_lexer_by_name
+
+from .constants import (
+    HEADER_SIZE, MAGIC_NUMBER, USED_BITS_MASK, USED_BITS_PER_BYTE
+)
+
+
+def make_image(code: str, *, language: str = "python", **formatter_options) -> PIL.Image.Image:
+    """
+    Generates image of the provided source code
+
+    :param code: source code, to be converted into image
+    :param language: programming language of source code
+    :param formatter_options: custom formatter options
+    :return: Image of source code
+    """
+    if not formatter_options:
+        # default formatter option
+        formatter_options = dict(
+            style="lightbulb",
+            line_number_bg=None,
+            line_number_separator=True,
+            line_number_pad=10,
+            image_pad=20,
+        )
+    lexer = get_lexer_by_name(language)
+    formatter = get_formatter_by_name("png", **formatter_options)
+    result = highlight(code, lexer, formatter)
+    return PIL.Image.open(BytesIO(result)).convert("RGBA")
+
+
+def chunk(data: Sequence[int]):
```

Comment:
the function name here could be improved as it returns chunks of data (a list). Also missing return type annotation.